### PR TITLE
Small asset editor cleanup, fixes

### PR DESCRIPTION
**Pull request by shakao**
Bundle of small fixes, I added comments to identify each. Also unifying the code for updating tiles when exiting the tilemap editor.

### CODE REVIEW

File: webapp/src/components/ImageEditor/tilemap/TilePalette.tsx [hunk 1]
```diff
@@ -487,7 +487,12 @@ class TilePaletteImpl extends React.Component<TilePaletteProps,{}> {
         return this.props.tileset.tiles
             .map((t, i) => ([t, i] as [pxt.Tile, number]))
             .filter(([t]) => t.isProjectTile)
-            .sort(([a], [b]) => a.internalID - b.internalID);
+            .sort(([a], [b]) => {
+                const transparency = "myTiles.transparency" + this.props.tileset.tileWidth;
```
Comment (shakao) [inline webapp/src/components/ImageEditor/tilemap/TilePalette.tsx:491]: transparency tile should always be drawn first in "My Tiles" palette view

File: webapp/src/components/assetEditor/editor.tsx [hunk 2]
```diff
@@ -101,6 +101,14 @@ export class AssetEditor extends Editor {
                 });
                 break;
             case pxt.AssetType.Tilemap:
+                // for tilemaps, fill in all project tiles
+                const allTiles = pxt.react.getTilemapProject().getProjectTiles(asset.data.tileset.tileWidth, true);
```
Comment (shakao) [inline webapp/src/components/assetEditor/editor.tsx:105]: move "fill code" out of individual "create"/"edit" handlers; always fill in all available tiles before opening the field view

File: webapp/src/components/assetEditor/editor.tsx [hunk 3]
```diff
@@ -112,7 +120,35 @@ export class AssetEditor extends Editor {
...
+        fieldView.onHide(() => {
+            const result = fieldView.getResult();
+            if (asset.type == pxt.AssetType.Tilemap) result.data = this.updateTilemapTiles(result.data);
+            cb(result);
+        });
         fieldView.show();
     }
+
+    protected updateTilemapTiles = (data: pxt.sprite.TilemapData): pxt.sprite.TilemapData => {
+        const project = pxt.react.getTilemapProject();
```
Comment (shakao) [inline webapp/src/components/assetEditor/editor.tsx:132]: move update of deleted and edited tiles out of individual "create"/"edit" callbacks, always update all tiles and trim tileset after exiting field editor

File: webapp/src/components/assetEditor/store/assetEditorReducer.ts [hunk 4]
```diff
@@ -56,9 +60,12 @@ function getUserAssets() {
         return asset;
     };
 
-    return project.getAssets(pxt.AssetType.Image).map(imageToGalleryItem)
-        .concat(project.getAssets(pxt.AssetType.Tile).map(imageToGalleryItem))
-        .concat(project.getAssets(pxt.AssetType.Tilemap).map(tilemapToGalleryItem));
+    const images = project.getAssets(pxt.AssetType.Image).map(imageToGalleryItem).sort(compareInternalId);
```
Comment (shakao) [inline webapp/src/components/assetEditor/store/assetEditorReducer.ts:63]: sort assets by internal id (newest created asset displayed last)

File: webapp/src/components/assetEditor/store/assetEditorReducer.ts [hunk 5]
```diff
@@ -56,9 +60,12 @@ function getUserAssets() {
...
+    const tiles = project.getAssets(pxt.AssetType.Tile).map(imageToGalleryItem)
+        .filter(t => !t.id.match(/^myTiles.transparency(8|16|32)$/gi)).sort(compareInternalId);
```
Comment (shakao) [inline webapp/src/components/assetEditor/store/assetEditorReducer.ts:65]: hide transparency tiles from asset editor view

**Review by jwunderl — APPROVED** [review]

**Comment by shakao** [comment]
dropped a commit! also add "assets.json" file if assetEditor is enabled in the target and the project does not yet have the file (backwards compat)

**Review by jwunderl — APPROVED** [review]

**Review by livcheerful — APPROVED** [review]

**Pull request merged**